src/widget/wsearchlineedit.h

```diff
@@ -11,16 +11,19 @@
 
 class SkinContext;
 
-class WSearchLineEdit : public QLineEdit, public WBaseWidget {
+class WSearchLineEdit : public QComboBox, public WBaseWidget {
     Q_OBJECT
   public:
     // Delay for triggering a search while typing
     static constexpr int kMinDebouncingTimeoutMillis = 100;
     static constexpr int kDefaultDebouncingTimeoutMillis = 300;
     static constexpr int kMaxDebouncingTimeoutMillis = 9999;
+    static constexpr int kSaveTimeoutMillis = 5000;
```

Comment:
This feels too long to me. I got confused why searches I typed weren't getting saved.

Reply:
I think the search is saved on focusOutEvent as well.
at least it got saved when I tried to search, then quickly Tab to the table, Tab back to Search (all in less than 5s)

Reply:
what I'm trying to say: what's your use case for saing the query when _not_ pressing Enter and when _not_ going to the tracks table?
if we bind saving the queries to the debouncing timeout we'd pollute the search history, right?

Reply:
yes. While testing I found 5 secs a good timestamp to for savepoints. I did not add a focus out handler to trigger the search, maybe some other code does that....

Reply:
@Be-ing you can always force a entry by pressing enter. if i have a search result shorter open then five seconds, it's not worth saving becaus i obviously moved on or did not even pick a track. In testing you might be faster, but not on normal usage. I don't want to have too much spam in the history, have baked search queries :)

Reply:
Okay, now that searches are saved with Enter I think this is fine.